### DISCUSSION TITLE
Way to access magic properties in Alpine.effect?

### DISCUSSION THREAD

**Discussion by wooooodward** [alpinejs/alpine#2618] [1. Help] [answered]
I'd like to access the magic properties `$nextTick` and `$refs` - is something like below possible when using `Alpine.store` and `Alpine.effect`?

```
<script>
  document.addEventListener('alpine:init', () => {
    Alpine.store('productFilters', {
      initial: true,
      modal: false,
      filters: {
        category: 'all',
        search: '',
      },

      ...
    })

    Alpine.effect(() => {
      const store = Alpine.store('productFilters')
      const search = store.filters.search
      const category = store.filters.category
      this.$nextTick(() => this.$refs.search.scrollIntoView())
    })
  })
</script>
```

A solution is to use `setTimeout` and to reference the element directly but then you're playing the guessing game at how long of a timeout is needed. I'd rather use `$nextTick` as it allows you to execute the function AFTER Alpine has made its reactive DOM updates.

**Comment by SimoTod** [comment, marked as answer]
No, magic properties are only available inside a component because of the way they get setup but you can use Alpine.nextTick which is basically the same thing and is available everywhere
